### PR TITLE
[WIP] Creating a new example with MEG data

### PR DESCRIPTION
**Pull request by ja-che**

### CODE REVIEW

File: hidimstat/data_preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        Rank reduction of the whitener. If None rank is estimated from data.
+
+    Returns
+    -------
+    G : array, shape (n_channels, n_dipoles)
+        The preprocessed gain matrix.
+
+    M : array, shape (n_channels, n_times)
+        The whitened MEG/EEG measurements.
+    """
```
Comment (bthirion) [inline hidimstat/data_preprocessing.py:46]: Maybe indicate in docstring that this is mostly a wrapper to MNE _prepare_gain.
the functiona name could contain `MNE`

Comment (agramfort) [reply hidimstat/data_preprocessing.py:46]: I would not put this in the lib. Just in an example. Keep this package generic. Same for fMRI you have preprocessing stuff in nilearn.

Comment (ja-che) [reply hidimstat/data_preprocessing.py:46]: Ok

File: examples/plot_meg_data_example.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+    stc = _make_sparse_stc(X, active_set, forward, tmin=evoked.times[0],
+                           tstep=1. / evoked.info['sfreq'])
+    return stc
+
+
+# Downloading data
+subject = 'sample'
+data_path = sample.data_path()
+fwd_fname = data_path + '/MEG/sample/sample_audvis-meg-eeg-oct-6-fwd.fif'
```
Comment (bthirion) [inline examples/plot_meg_data_example.py:93]: you should use `os.path.join`

File: doc/conf.py [hunk 3]
```diff
@@ -204,13 +205,56 @@
     'joblib': ('https://joblib.readthedocs.io/en/latest', None),
     'pandas': ('https://pandas.pydata.org/pandas-docs/stable', None),
     'seaborn': ('https://seaborn.pydata.org/', None),
+    'mayavi': ('http://docs.enthought.com/mayavi/mayavi', None),
```
Comment (agramfort) [inline doc/conf.py:208]: remove mayavi

File: doc/conf.py [hunk 4]
```diff
@@ -204,13 +205,56 @@
...
+    # Do not pop up any mayavi windows while running the
+    # examples. These are very annoying since they steal the focus.
+    mlab.options.offscreen = True
+    # hack to initialize the Mayavi Engine
+    mlab.test_plot3d()
+    mlab.close()
+except Exception:
+    pass
+else:
+    scrapers += ('mayavi',)
```
Comment (agramfort) [inline doc/conf.py:228]: remove this. No need for mayavi

File: doc/conf.py [hunk 5]
```diff
@@ -204,13 +205,56 @@
...
+try:
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=DeprecationWarning)
+        import pyvista
+    pyvista.OFF_SCREEN = False
+except Exception:
+    pass
+else:
+    scrapers += ('pyvista',)
+if any(x in scrapers for x in ('pyvista', 'mayavi')):
```
Comment (agramfort) [inline doc/conf.py:238]: remove mayavi

File: doc/doc-requirements.txt [hunk 6]
```diff
@@ -8,3 +8,6 @@ scikit-learn
 scipy
 sphinx-bootstrap-theme
 sphinx-gallery
+mne
+mayavi
```
Comment (agramfort) [inline doc/doc-requirements.txt:12]: remove mayavi

**Pull request merged**